huaweicloud/data_source_huaweicloud_networking_secgroup.go

```diff
@@ -53,48 +56,110 @@ func DataSourceNetworkingSecGroup() *schema.Resource {
 	}
 }
 
+func getSecurityGroupByV1Client(d *schema.ResourceData, config *config.Config) (*v1Groups.SecurityGroup, error) {
+	v1Client, err := config.SecurityGroupV1Client(GetRegion(d, config))
+	if err != nil {
+		return nil, fmtp.Errorf("Error creating HuaweiCloud networking v1 client: %s", err)
+	}
+
+	listOpts := v1Groups.ListOpts{
+		EnterpriseProjectId: config.DataGetEnterpriseProjectID(d),
+	}
+
+	pages, err := v1Groups.List(v1Client, listOpts).AllPages()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get security groups: %s", err)
+	}
+	allGroups, err := v1Groups.ExtractSecurityGroups(pages)
+	if err != nil {
+		return nil, fmt.Errorf("Error retriving security group list: %s", err)
+	}
+	if len(allGroups) == 0 {
+		return nil, fmt.Errorf("No sucurity group found, please change your search criteria and try again.")
+	}
+
+	filter := map[string]interface{}{
+		"ID":   d.Get("secgroup_id"),
+		"Name": d.Get("name"),
+	}
+	filterGroups, err := utils.FilterSliceWithField(allGroups, filter)
+	if err != nil {
+		return nil, fmtp.Errorf("Erroring filting security groups: %s", err)
+	}
+	if len(filterGroups) < 1 {
+		return nil, fmtp.Errorf("No Security Group found.")
+	}
+	if len(filterGroups) > 1 {
+		return nil, fmtp.Errorf("More than one Security Groups found.")
+	}
+
+	result := filterGroups[0].(v1Groups.SecurityGroup)
+	return &result, nil
+}
+
+func setV1SecurityGroupDetail(d *schema.ResourceData, config *config.Config, region string) diag.Diagnostics {
+	resp, err := getSecurityGroupByV1Client(d, config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	rules, err := flattenV1SecurityGroupRules(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("name", resp.Name),
+		d.Set("description", resp.Description),
+		d.Set("enterprise_project_id", resp.EnterpriseProjectId),
+		d.Set("rules", rules),
```

Comment:
the setting fields are different between v3 and v1

Reply:
In the ver.1 response, the structure does not have `created_at` and `updated_at`, but it has `enterprise_project_id`, this parameter that was missing in the old code's settings.